### PR TITLE
✨ Make user agent request use friendly app version

### PR DESCRIPTION
**Pull request by Sven65**
## Explain the changes you’ve made

In accordance with #CU-1w4yn9z, I've changed the app version in `ApiRequest.js` to use the `getFriendlyAppVersion` function

## Explain your solution

I'm simply using the `getFriendlyAppVersion` function instead of `version` from `package.json`

## How to test

Concrete example:

1. Checkout this branch
2. Fire up the simulator with `yarn ios`
3. Fire up devtools that are capable of inspecting network requests, I recommend [React Native Debugger](https://github.com/jhen0409/react-native-debugger)
4. Send any request, i.e logging in, fetch cases etc
5. See that the `User-Agent` header is using the friendly app version instead

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1w4yn9z Ändra version i user-agent från package.json till samma som används på loginskärmen](https://app.clickup.com/t/1w4yn9z)

**Review by jlarsson — APPROVED** [review]
Looks good.

File: source/helpers/ApiRequest.js [hunk 1]
```diff
@@ -19,7 +20,10 @@ const request = async (endpoint, method, data, headers, params) => {
   const token = await StorageService.getData(ACCESS_TOKEN_KEY);
   const { apiKey } =
     EnvironmentConfigurationService.getInstance().activeEndpoint;
-  const userAgent = `${name}/${version}/${Platform.OS}/${Platform.Version}`;
+
+  const friendlyVersion = getUserFriendlyAppVersion();
+
+  const userAgent = `${name}/${friendlyVersion}/${Platform.OS}/${Platform.Version}`;
```
Comment (karlknowit) [inline source/helpers/ApiRequest.js:26]: This change is creating an `UserAgent` that looks like the following:

`MittHelsingborg/v3.1-15-53025bb/ios/15.2`

Which means that we can no longer distinguish the current app version (in backend).
The **old** `UserAgent` on the other hand looked like the following:

`MittHelsingborg/1.1.6/ios/15.2`
where `1.1.6` is the current app version used.

So we either prevent this to be merged or come up with another solution regarding the `UserAgent` and specifically current application version.

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/helpers/ApiRequest.js (hunk 1 above)
Comment (karlknowit) [reply source/helpers/ApiRequest.js:26]: Change `UserAgent` to `MittHelsingborg/1.1.6/ios/15.2/v.3.1-15-53025bb` as agreed. By doing this change we won't interfere with current code in backend. We might want another `UserAgent` in the future, but that might need discussions first.

**Review by karlknowit — CHANGES_REQUESTED** [review]

**Review by karlknowit — APPROVED** [review]

**Pull request merged**